src/main/java/org/candlepin/dto/api/v1/ConsumerTranslator.java

```diff
@@ -137,8 +138,16 @@ public ConsumerDTO populate(ModelTranslator translator, Consumer source, Consume
                 dest.setOwner(owner != null ? translator.translate(owner, NestedOwnerDTO.class) : null);
             }
 
-            Environment environment = this.environmentCurator.getConsumerEnvironment(source);
-            dest.setEnvironment(translator.translate(environment, EnvironmentDTO.class));
+            if (source.getEnvironmentIds() != null) {
+                Collection<String> envIDs = source.getEnvironmentIds().values();
+                for (String id : envIDs) {
+                    dest.addEnvironments(translator
+                        .translate(this.environmentCurator.listById(id), EnvironmentDTO.class));
```

Comment:
This needs to be done with a single query and translation, as we're already leaning into N+1 efficiency issues.

Reply:
Yeah, gathered all environments and later translated.